server/plugin.go

```diff
@@ -138,6 +161,157 @@ func (p *Plugin) handleList(w http.ResponseWriter, r *http.Request) {
 	w.Write(itemsJSON)
 }
 
+type enqueueAPIRequest struct {
+	ID string
+}
+
+func (p *Plugin) handleEnqueue(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	var enqueueRequest *enqueueAPIRequest
+	decoder := json.NewDecoder(r.Body)
+	err := decoder.Decode(&enqueueRequest)
+	if err != nil {
+		p.API.LogError("Unable to decode JSON err=" + err.Error())
+		w.WriteHeader(http.StatusBadRequest)
```

Comment:
This may already be the convention in the TODO plugin, but generally we should return JSON errors.

Reply:
Not sure what you meant here. Can you elaborate?

Reply:
Take a look at how workflow or jira returns errors -- as json objects, so that the front end can decode them and display them to the user in the UI.